diff --git a/gedit/gedit-bottom-panel.c b/gedit/gedit-bottom-panel.c
index 75196a3be..3954bc4b0 100644
--- a/gedit/gedit-bottom-panel.c
+++ b/gedit/gedit-bottom-panel.c
@@ -67,7 +67,6 @@ _gedit_bottom_panel_init (GeditBottomPanel *panel)
 			 0, 0, 1, 1);
 
 	switcher = gedit_notebook_stack_switcher_new ();
-	gtk_widget_set_vexpand (switcher, TRUE);
 	gedit_notebook_stack_switcher_set_stack (GEDIT_NOTEBOOK_STACK_SWITCHER (switcher),
 						 panel->priv->stack);
 	gtk_widget_show (switcher);
